### ISSUE TITLE
Workaround MySQL bug 79533

### ISSUE THREAD

**Issue by SuperQ** [prometheus/mysqld_exporter#127]
It turns out that MySQL sometimes duplicates rows in `events_statements_summary_by_digest`

Possibly in response to TRUNCATE TABLE

```
> select DIGEST, SCHEMA_NAME, COUNT(*), DIGEST_TEXT FROM events_statements_summary_by_digest group by DIGEST, SCHEMA_NAME HAVING COUNT(*) > 1;
+----------------------------------+-----------------------+----------+--------------------------------------------------------------------+
| DIGEST                           | SCHEMA_NAME           | COUNT(*) | DIGEST_TEXT                                                        |
+----------------------------------+-----------------------+----------+--------------------------------------------------------------------+
| a9d847b1b646cdc5781d41422ffb4aa2 | soundcloud_production |        2 | BEGIN                                                              |
```

See: https://bugs.mysql.com/bug.php?id=79533

**Comment by beorn7** [comment]
The bug is fixed upstream, but mysqld_exporter should be able to defend against this and filter out the duplicates proactively (rather than letting them hit the Prometheus server, where they will be discarded anyway but trigger warnings).

Quite possible that newer client_golang versions will error out on this anyway (but they can be configured to still expose metrics, filtering out the duplicates).

**Comment by beorn7** [comment]
Hmm, it appears they have closed the bug but didn't fix it. Declared it a feature, essentially...
Whatever, we need to defend against this, as said above.

**Comment by SuperQ** [comment]
That's amazing, MySQL++ </sarcasm>

**Comment by SuperQ** [comment]
Fixed with #173
